src/Faction.js

```diff
@@ -5,15 +5,35 @@ class Faction {
   }
 
   static fromCode (code) {
-    if (!Faction.FACTIONS.includes(code)) throw new TypeError('Invalid faction code!')
-    return new this(code, this.FACTIONS.indexOf(code))
+    const factionId = Faction.FACTIONS[code]
+
+    if (factionId === undefined) {
+      throw new TypeError('Invalid faction code. It is possible you need to upgrade the runeterra package.')
+    }
+
+    return new this(code, factionId)
   }
 
   static fromID (id) {
-    return new this(this.FACTIONS[id], id)
+    const [shortCode, factionId] = Object.entries(Faction.FACTIONS).find(([shortCode, factionId]) => factionId === id) || []
+
+    if (factionId === undefined) {
+      throw new TypeError('Invalid faction id. It is possible you need to upgrade the runeterra package.')
```

Comment:
Added an error here as well. This should also be failing for the same reason as above because if not it will return a completely bogus faction that doesn't make sense.